### PR TITLE
Upgrade 3.3

### PR DESCRIPTION
**Pull request by cjay**
Making this compatible with https://github.com/bsl/bindings-GLFW/pull/65 so it can compile. I suggest merging both at the same time, so the Upgrade-3.3 branches of both repos stay compatible. Not added new functions yet.

### CODE REVIEW

**Comment by cjay** [comment]
Caution: Had to do this request against bsl:master, the original repo doesn't have an Upgrade-3.3 branch yet.

**Comment by Mokosha** [comment]
Hi -- Thanks for your PR. I think we're waiting until 3.3 gets released before we do anything on master. If you'd like to contribute to Upgrade-3.3 in the meantime, I'm happy to take PRs there!

**Comment by cjay** [comment]
It seems you have to create the Upgrade-3.3 branch first, github won't let me select a branch that doesn't exist yet

**Comment by Mokosha** [comment]
Thanks for pointing that out! I've created the `Upgrade-3.3` branch and deleted `Upgrade-3.2.1` which has been fully merged. We can add TODOs to flesh it out once 3.3 is released but feel free to send PRs against it in the meantime.

**Comment by cjay** [comment]
Ok I changed the base branch to Upgrade-3.3.

File: GLFW-b.cabal [hunk 1]
```diff
@@ -16,14 +16,14 @@ description:
   library for creating windows with OpenGL contexts and managing input and
   events.
   .
-  GLFW-b depends on bindings-GLFW
+  This version of GLFW-b depends on git branch Upgrade-3.3 of bindings-GLFW
   (<http://hackage.haskell.org/package/bindings-GLFW>), which, as of the time
-  of this writing, binds to GLFW 3.2.1, released 2016-08-18
-  (<http://www.glfw.org/Version-3.2.1-released.html>
+  of this writing, binds to GLFW 3.3, not yet released
```
Comment (Mokosha) [inline GLFW-b.cabal]: It was released on 2019-04-15

File: GLFW-b.cabal [hunk 2]
```diff
@@ -16,14 +16,14 @@ description:
   library for creating windows with OpenGL contexts and managing input and
   events.
   .
-  GLFW-b depends on bindings-GLFW
+  This version of GLFW-b depends on git branch Upgrade-3.3 of bindings-GLFW
```
Comment (Mokosha) [inline GLFW-b.cabal]: I would suggest we start work on upgrading `bindings-GLFW` first before changing this text.

**Comment by cjay** [comment]
Ok I reverted the changes to the cabal file. Please don't forget about https://github.com/bsl/bindings-GLFW/pull/65, both are currently needed for Vulkan on macOS as I described [here](https://github.com/achirkin/vulkan/blob/master/README-macOS.md#installing-the-needed-haskell-libraries).

**Pull request merged**